torch/csrc/autograd/python_function.cpp

```diff
@@ -607,9 +607,14 @@ PyObject* THPFunction_name(PyObject *self, PyObject* noargs) {
 PyObject *THPFunction_apply(PyObject *cls, PyObject *inputs)
 {
   HANDLE_TH_ERRORS
+  auto info_pair = unpack_input<false>(inputs);
+  auto iv_inputs = std::vector<c10::IValue>();
+  for (auto i : info_pair.first.input_vars) {
```

Comment:
This is going to be pretty wasteful in the case where we are not recording.
Is there a way to skip that work in such cases?

Reply:
Maybe add a check like in the dispatcher: https://github.com/pytorch/pytorch/blob/25e2578967494ee88da66820d861c426eb7a742d/aten/src/ATen/core/dispatch/Dispatcher.h#L550

Reply:
Check added.

Reply:
There's also a check `if (guard.needsInputs())` (see pytorch/aten/src/ATen/core/dispatch/Dispatcher.h)

Also `auto info_pair = unpack_input<false>(inputs);` is also expensive and can be put inside the conditional.

Reply:
We need the info_pair for later in that function.

Reply:
> We need the info_pair for later in that function.

I see, makes sense.